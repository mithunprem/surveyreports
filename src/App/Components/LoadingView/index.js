import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Spinner from '../Spinner';
import { Alert } from 'reactstrap';

const LoadingView = ({ isLoading, text }) => {
  return (
    <Fragment>
      {
        isLoading ?
          <Spinner spinnerText={`Loading ${text}â€¦`} /> :
          <Alert color="danger">
            {`Failed to load ${text}. Please try again !.`}
          </Alert>
      }
    </Fragment>
  )
}

LoadingView.propTypes = {
  isLoading: PropTypes.bool,
  text: PropTypes.string
}

export default LoadingView;
