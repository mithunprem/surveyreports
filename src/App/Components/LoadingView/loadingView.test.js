import React from 'react';
import { create } from 'react-test-renderer';
import LoadingView from '../../Components/LoadingView';

describe("LoadingView", () => {
  test("should render a loading spinner without throwing an error", () => {
    const loadingView = create(<LoadingView isLoading={true} text={'test'}/>).toJSON();
    expect(loadingView).toMatchSnapshot();
  });

  test("should render an alert view without throwing an error", () => {
    const loadingView = create(<LoadingView isLoading={false} text={'test'}/>).toJSON();
    expect(loadingView).toMatchSnapshot();
  });
});
