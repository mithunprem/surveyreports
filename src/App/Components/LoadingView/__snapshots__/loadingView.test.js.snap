// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingView should render a loading spinner without throwing an error 1`] = `
<div
  className="m-4"
>
  Loading testâ€¦
   
  <div
    className="spinner-grow text-dark"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`LoadingView should render an alert view without throwing an error 1`] = `
<div
  className="alert alert-danger fade"
  role="alert"
>
  Failed to load test. Please try again !.
</div>
`;
